src/garage/torch/algos/maml/maml.py

```diff
@@ -0,0 +1,478 @@
+
+            update_module_params(self._policy, theta)
+            update_module_params(self._old_policy, old_theta)
+
+        return torch.stack(kls).mean()
+
+    def _compute_loss(self, itr, batch_samples):
+        """Compute loss for one task and one gradient step.
+
+        Args:
+            itr (int): Iteration number.
+            batch_samples (MAMLTrajectoryBatch): Samples data for one
+                task and one gradient step.
+
+        Returns:
+            torch.Tensor: Computed loss value.
+
+        """
+        # pylint: disable=protected-access
+        return self._inner_algo._compute_loss(itr, batch_samples.observations,
+                                              batch_samples.actions,
+                                              batch_samples.rewards,
+                                              batch_samples.lengths,
+                                              batch_samples.baselines)
+
+    def _compute_policy_entropy(self, task_samples):
+        """Compute policy entropy.
+
+        Args:
+            task_samples (list[MAMLTrajectoryBatch]): Samples data for
+                one task.
+
+        Returns:
+            torch.Tensor: Computed entropy value.
+
+        """
+        # pylint: disable=protected-access
+        entropies = [
+            self._inner_algo._compute_policy_entropy(samples.observations)
+            for samples in task_samples
+        ]
+        return torch.stack(entropies).mean()
+
+    def _set_task(self, runner, task):
+        # pylint: disable=protected-access, no-self-use
+        for env in runner._sampler._vec_env.envs:
+            env.set_task(task)
+
+    @property
+    def policy(self):
+        """Current policy of the inner algorithm.
+
+        Returns:
+            garage.torch.policies.Policy: Current policy of the inner
+                algorithm.
+
+        """
+        return self._policy
+
+    @property
+    def _old_policy(self):
+        """Old policy of the inner algorithm.
+
+        Returns:
+            garage.torch.policies.Policy: Old policy of the inner algorithm.
+
+        """
+        # pylint: disable=protected-access
+        return self._inner_algo._old_policy
+
+    def _process_samples(self, itr, paths):
+        """Process sample data based on the collected paths.
+
+        Args:
+            itr (int): Iteration number.
+            paths (list[dict]): A list of collected paths
+
+        Returns:
+            MAMLTrajectoryBatch: Processed samples data.
+
+        """
+        for path in paths:
+            path['returns'] = tensor_utils.discount_cumsum(
+                path['rewards'], self._inner_algo.discount)
+
+        self._baseline.fit(paths)
+        obs, actions, rewards, valids, baselines \
+            = self._inner_algo.process_samples(itr, paths)
+        return MAMLTrajectoryBatch(obs, actions, rewards, valids, baselines)
+
+    def evaluate_performance(self, itr, all_samples, loss_before, loss_after,
```

Comment:
it is essential that the performance evaluation part of the algorithms is shared. 

i would prefer if this called `RLAlgorithm.evaluate_performance` instead. Alternatively, it may be necessary to introduce `MetaRLAlgorithm` with `evaluate_train_performance` and `evaluate_test_performance`